Subject: ethtool: Remove support for RX hashing by port
From: Ben Hutchings <bhutchings@solarflare.com>
Date:	Thu, 30 Apr 2009 13:36:49 +0100

The kernel-side support for this was removed by:

commit 59089d8d162ddcb5c434672e915331964d38a754
Author: Santwona Behera <santwona.behera@sun.com>
Date:   Fri Feb 20 00:58:13 2009 -0800

    ethtool: Add RX pkt classification interface

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
---
 ethtool.8 |    3 ---
 ethtool.c |    8 +-------
 2 files changed, 1 insertions(+), 10 deletions(-)

diff --git a/ethtool.8 b/ethtool.8
index 178f6ea..7a6f52b 100644
--- a/ethtool.8
+++ b/ethtool.8
@@ -473,9 +473,6 @@ Configures the hash options for the specified network traffic type.
 .RS
 .PD 0
 .TP 3
-.B p
-Hash on the device port number on which the packet was received.
-.TP 3
 .B m
 Hash on the Layer 2 destination address of the rx packet.
 .TP 3
diff --git a/ethtool.c b/ethtool.c
index 43ec2bf..5c45b2d 100644
--- a/ethtool.c
+++ b/ethtool.c
@@ -191,7 +191,7 @@ static struct option {
     { "-N", "--config-nfc", MODE_SNFC, "Configure Rx network flow "
 		"classification options",
 		"		[ rx-flow-hash tcp4|udp4|ah4|sctp4|"
-		"tcp6|udp6|ah6|sctp6 p|m|v|t|s|d|f|n|r... ]\n" },
+		"tcp6|udp6|ah6|sctp6 m|v|t|s|d|f|n|r... ]\n" },
     { "-h", "--help", MODE_HELP, "Show this help" },
     {}
 };
@@ -1090,9 +1090,6 @@ static int parse_rxfhashopts(char *optstr, u32 *data)
 	*data = 0;
 	while (*optstr) {
 		switch (*optstr) {
-			case 'p':
-				*data |= RXH_DEV_PORT;
-				break;
 			case 'm':
 				*data |= RXH_L2DA;
 				break;
@@ -1132,9 +1129,6 @@ static char *unparse_rxfhashopts(u64 opts)
 	memset(buf, 0, sizeof(buf));
 
 	if (opts) {
-		if (opts & RXH_DEV_PORT) {
-			strcat(buf, "Dev port\n");
-		}
 		if (opts & RXH_L2DA) {
 			strcat(buf, "L2DA\n");
 		}
